documentation/internal/Element-Type-Specs.md

```diff
@@ -152,6 +152,47 @@ _Multi-choice question, with one allowed option, displayed as Drop-down list (Co
 
 ---
 
+### Checkboxes
+
+- **type/code**: `checkbox`
+- **category**: `Question`
+
+_One or more checkboxes, any number of which can be selected/toggled_
+
+#### Input parameters
+
+- **label\***: `string` -- as above
+- **description\***: `string` -- as above [Optional]
+- **checkboxes\***: `array[string | checkbox]` -- an array of labels, one per checkbox. For more complexity, an array of Checkbox objects can be provided with the following properties:
+
+  ```
+   {
+     label: <string> - text to display next to checkbox (Can be empty string but not omitted)
+     text: <string> - value to store in Response "text" field and shown in Summary View. Will be same as label if omitted.
+     key: <string | number> - unique code used as key/property name for Response object. Defaults to numerical index of array if omitted
+     selected: <boolean> - initial state of checkbox
+   }
+  ```
```

Comment:
All good here. I just wonder if for an extension of this plugin we would have here another option to enter something to link this option to e.g. a `permission_name` or an action.

Reply:
Yes, we still have to figure out exactly how we're going to link these things (using the "reference" field we've alluded to in the docs but never really fleshed out). But the response object for any plugin can easily be extended at any time to include additional fields. 

I've been bearing it in mind, but I think we won't know *exactly* how it's going to connect back to other entities until we go to implement one -- probably the "select a company to join" option will be the first.